cmd/server-main.go

```diff
@@ -238,21 +238,25 @@ func getListenIPs(httpServerConf *http.Server) (hosts []string, port string) {
 
 // Finalizes the endpoints based on the host list and port.
 func finalizeEndpoints(tls bool, apiServer *http.Server) (endPoints []string) {
-	// Get list of listen ips and port.
-	hosts, port := getListenIPs(apiServer)
-
 	// Verify current scheme.
 	scheme := "http"
 	if tls {
 		scheme = "https"
 	}
 
+	// Get list of listen ips and port.
+	hosts, port := getListenIPs(apiServer)
 	ips := getIPsFromHosts(hosts)
-
 	// Construct proper endpoints.
 	for _, ip := range ips {
 		endPoints = append(endPoints, fmt.Sprintf("%s://%s:%s", scheme, ip.String(), port))
 	}
+	// If hosts is empty, add more endpoints.
+	for _, host := range hosts {
```

Comment:
Yes when someone has specified a hostname along with the port in the "address" argument.. getListenIPs() would return an empty "ips" when that is done since it is meant to be used only for ":9000" type of inputs.